tunnel_virtual_networks.go

```diff
@@ -0,0 +1,157 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// TunnelVirtualNetwork is segregation of Tunnel IP Routes via Virtualized Networks to handle overlapping private IPs in your origins.
+type TunnelVirtualNetwork struct {
+	ID               string     `json:"id"`
+	Name             string     `json:"name"`
+	IsDefaultNetwork bool       `json:"is_default_network"`
+	Comment          string     `json:"comment"`
+	CreatedAt        *time.Time `json:"created_at"`
+	DeletedAt        *time.Time `json:"deleted_at"`
+}
+
+type TunnelVirtualNetworksListParams struct {
+	AccountID string
+	ID        string `url:"id,omitempty"`
+	Name      string `url:"name,omitempty"`
+	IsDefault *bool  `url:"is_default,omitempty"`
+	IsDeleted *bool  `url:"is_deleted,omitempty"`
+	PaginationOptions
+}
+
+type TunnelVirtualNetworkCreateParams struct {
+	AccountID string `json:"-"`
+	VnetName  string `json:"vnet_name"`
+	Comment   string `json:"comment,omitempty"`
+	IsDefault *bool  `json:"is_default,omitempty"`
+}
+
+type TunnelVirtualNetworkUpdateParams struct {
+	AccountID        string `json:"-"`
+	VnetID           string `json:"-"`
+	VnetName         string `json:"vnet_name,omitempty"`
+	Comment          string `json:"comment,omitempty"`
+	IsDefaultNetwork *bool  `json:"is_default_network,omitempty"`
+}
+
+type TunnelVirtualNetworkDeleteParams struct {
+	AccountID string `json:"-"`
+	VnetID    string `json:"-"`
+}
+
+// tunnelRouteListResponse is the API response for listing tunnel virtual networks.
+type tunnelVirtualNetworkListResponse struct {
+	Response
+	Result []TunnelVirtualNetwork `json:"result"`
+}
+
+type tunnelVirtualNetworkResponse struct {
+	Response
+	Result TunnelVirtualNetwork `json:"result"`
+}
+
+// ListTunnelVirtualNetworks lists all defined virtual networks for tunnels in the account.
+//
+// See: https://api.cloudflare.com/#tunnel-virtual-network-list-virtual-networks
```

Comment:
this will need to be updated on all the methods in this file

```suggestion
// API documentation: https://api.cloudflare.com/#tunnel-virtual-network-list-virtual-networks
```